src/main/java/dk/cngroup/lentils/controller/ScoreController.java

```diff
@@ -0,0 +1,55 @@
+package dk.cngroup.lentils.controller;
+
+import dk.cngroup.lentils.entity.Cypher;
+import dk.cngroup.lentils.entity.CypherStatus;
+import dk.cngroup.lentils.entity.Hint;
+import dk.cngroup.lentils.entity.Team;
+import dk.cngroup.lentils.factory.CypherStatusFactory;
+import dk.cngroup.lentils.service.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+@Controller
+@RequestMapping("/game/score")
+public class ScoreController {
+    private static final String PROGRESS_STAGE = "progress/stage";
+    private static final String SCORE_LIST = "score/list";
+    private static final String ERROR = "error/error";
+    private static final String HINT_LIST = "progress/getHintList";
+
+    private final TeamService teamService;
+    private final CypherService cypherService;
+    private final ProgressService progressService;
+    private final HintService hintService;
+    private final HintTakenService hintTakenService;
+    private final ScoreService scoreService;
+
+    @Autowired
+    public ScoreController(final TeamService teamService,
+                           final CypherService cypherService,
+                           final ProgressService progressService,
+                           final HintService hintService,
+                           final HintTakenService hintTakenService,
+                           final ScoreService scoreService) {
+        this.teamService = teamService;
+        this.cypherService = cypherService;
+        this.progressService = progressService;
+        this.hintService = hintService;
+        this.hintTakenService = hintTakenService;
+        this.scoreService = scoreService;
+    }
+
+    @GetMapping
+    public String listScore(final Model model) {
+        model.addAttribute("teamsAndScores", scoreService.getAllTeamsWithScores());
+        return SCORE_LIST;
+    }
+
+    @ExceptionHandler(IllegalArgumentException.class)
```

Comment:
Idealni by bylo, kdyby tady tato metoda vubec nevyla. Pokud se nekde vyhazuje illegal argument exception pak by se to mohlo udelat tak, ze se vytvori nova vyjimka (nase), ktera rozsiruje illegal argument exception a primo v ni se definuje, kam se to ma presmerovat, kdyz se vyhodi. Tak jak je to napriklad v MoreFinalPlacesException.

Reply:
vyhodila jsem ji, zůstala mi tam z nějakého jiného controlleru, když jsem začínala a asi není potřeba